### PR TITLE
Changelog entry for v0.3.0

### PR DESCRIPTION
**Pull request by willschlitzer**
**Description of proposed changes**

**Preview**: https://pygmt-git-v030-release.gmt.vercel.app/changes.html

Previous release at #683

Changelog made by following the instructions at https://github.com/GenericMappingTools/pygmt/blob/master/MAINTENANCE.md#updating-the-changelog. The changelog itself includes the following sections:

- New features
- Enhancements
- Documentation
- Bug fixes
- Maintenance
- List of Contributors

TODO: 

- [x] Format Pull Request links using ``sed -i.bak -E 's$\(#([0-9]*)\)$(`#\1 <https://github.com/GenericMappingTools/pygmt/pull/\1>`__)$g' changes.rst``
- [x] Add Zenodo DOI link once it's reserved.
- [x] Add a new entry in doc/_static/version_switch.js for documentation switcher
- [x] Update citation information https://github.com/GenericMappingTools/pygmt#citing-pygmt
- [x] Set release date

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: README.rst
Comment (weiji14) [inline README.rst:174]: Please add a v0.3.0 entry under the 'Documentation for other versions' and 'Compatibility with Python and GMT versions' below.

**Review by weiji14 — COMMENTED** [review]
Hi @willschlitzer, assuming that you're learning from the previous release PR at #683 already. if you don't mind, could you edit the top post to follow that style (with checkboxes and all). We'll probably turn it into a proper template for #885 next time.

image

**Comment by weiji14** [comment]
Oh and by the way, I see that you've cleaned up the draft release notes at https://github.com/GenericMappingTools/pygmt/releases. Make sure you have a local **backup**, otherwise the release drafter bot will override your changes when we merge the last few PRs remaining for v0.3.0.

**Comment by willschlitzer** [comment]
@weiji14 It'S backed up!

File: README.rst
Comment (willschlitzer) [reply README.rst:174]: Updated in 1081fb2

File: README.rst
Comment (seisman) [inline README.rst:233]: 1. See https://github.com/GenericMappingTools/pygmt/pull/763#discussion_r549190174 for why we set `>=3.6` for PyGMT v0.2.1
2. The [minimum required Python version is 3.7](https://github.com/GenericMappingTools/pygmt/blob/02858eb5f916d3071ab2b104e41e4e8251896efb/setup.py#L30) for PyGMT v0.3.0
```suggestion
0.3.0   >=6.1.1    >=3.7
0.2.1   >=6.1.1    >=3.6
```

File: README.rst
Comment (weiji14) [inline README.rst:164]:
```suggestion
                      Ziebarth, Malte and
                      Jones, Meghan and
```

Need to add Malte too.

And just checking with @meghanrjones can I put down you and Paul's affiliation as "Department of Earth Sciences, School of Ocean and Earth Science and Technology, University of Hawai‘i at Mānoa" in Zenodo?

File: README.rst
Comment (weiji14) [inline README.rst:162]: Guten tag @michaelgrund, could you please provide your full affiliation (or would you prefer to use Unaffiliated)? Preferably in the format: Company, City, Country.

Better yet if you do a quick Pull Request to update it in https://github.com/GenericMappingTools/pygmt/blame/5faf06c30f2861e92e8bf8c7377451b95baeaec0/AUTHORS.md#L15. Just need it for the Zenodo DOI later.

File: doc/changes.rst
Comment (weiji14) [inline doc/changes.rst:13]: We'll need to add 3 to 4 lines for the highlights section here. How about:

```suggestion
* 🎉 **Third minor release of PyGMT** 🎉
* Wrap inset (#788) for making overview maps and subplot (#822) for multi-panel figures
* Apply standardized formatting conventions (#775) across most documentation pages
* Drop Python 3.6 support (#699) so PyGMT now requires Python 3.7 or newer
```
Feel free to suggest edits in the wording. I'm also thinking whether to mention the upcoming drop in support for GMT 6.1.1 (once GMT 6.2.0 is out, something for PyGMT v0.4.0) and/or NEP29 adoption (#690). Might depend on whether we want a v0.3.1 release?

File: doc/changes.rst
Comment (weiji14) [inline doc/changes.rst:105]:
```suggestion
People with a "+" by their name contributed a patch for the first time.

* `Dongdong Tian <https://github.com/seisman>`__
* `Liam Toney <https://github.com/liamtoney>`__
* `Meghan Jones <https://github.com/meghanrjones>`__ +
* `Michael Grund <https://github.com/michaelgrund>`__ +
* `Wei Ji Leong <https://github.com/weiji14>`__
* `Will Schlitzer <https://github.com/willschlitzer>`__ +
```

Two thoughts here:
1. Should we sort alphebetically by name (as above), or by number of commits (old style before).
2. Some projects (e.g. [geopandas](https://github.com/geopandas/geopandas/releases/tag/v0.8.0)) add a little note with a plus (+) to indicate people's first contribution, should we add it here?

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:19]:
```suggestion
* Provide basic support for FreeBSD (#700, #878)
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:24]:
```suggestion
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:23]:
```suggestion
* Let load_earth_relief() support the 'region' parameter for all resolutions (#873)
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:27]:
```suggestion
* Add the shading parameter to Figure.colorbar (#752)
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:37]:
```suggestion
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:40]:
```suggestion
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:48]:
```suggestion
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:55]:
```suggestion
* Add more gallery examples for projections (#761, #721, #757, #723, #762, #742)
```

File: README.rst
Comment (meghanrjones) [reply README.rst:164]: Yes, sounds good.

File: doc/changes.rst
Comment (seisman) [reply doc/changes.rst:105]: > Two thoughts here:
> 
> 1. Should we sort alphebetically by name (as above), or by number of commits (old style before).

I'm fine with both, but prefer "number of commits".

> 2. Some projects (e.g. [geopandas](https://github.com/geopandas/geopandas/releases/tag/v0.8.0)) add a little note with a plus (+) to indicate people's first contribution, should we add it here?

I saw it in many projects, but is there an easy way to know if someone makes his/her first contribution to this release?

File: doc/changes.rst
Comment (weiji14) [reply doc/changes.rst:37]: I'll add a skip-changelog label to this PR.

File: doc/changes.rst
Comment (weiji14) [reply doc/changes.rst:40]: Same, I'll add a skip-changelog label to this PR (nothing against Liam).

File: doc/changes.rst
Comment (weiji14) [reply doc/changes.rst:48]: Assuming that we just use the line at https://github.com/GenericMappingTools/pygmt/pull/887/files#r575914421? If so, then we should add skip-changelog tags to all of these.

Comment (seisman) [reply doc/changes.rst]: Do we really need to add "skip-changelog" label to this PRs? The "skip-changelog" label won't have any effects for future release notes, right?

Comment (weiji14) [reply doc/changes.rst]: Mm yes. It's more just in case we get another PR merged in before this changelog PR is finalized and merged.

Comment (seisman) [reply doc/changes.rst]: Even though more PRs are merged before we finalize this changelog, we won't go to the "draft release notes" and start everything again. So labeling "skip-changelog" now seems useless to me.

File: doc/changes.rst
Comment (weiji14) [reply doc/changes.rst:101]: > > 1. Some projects (e.g. [geopandas](https://github.com/geopandas/geopandas/releases/tag/v0.8.0)) add a little note with a plus (+) to indicate people's first contribution, should we add it here?
> 
> I saw it in many projects, but is there an easy way to know if someone makes his/her first contribution to this release?

Other than remembering people, there's probably a complicated git command that could do it. Happy to leave it without the plus.

> I'm fine with both, but prefer "number of commits".

I thought your name would be on top either way, but `git shortlog HEAD...v0.2.1 -sne` gives:

```
    58  Will Schlitzer <schlitzer90@gmail.com>
    54  Dongdong Tian <seisman.info@gmail.com>
     9  Wei Ji <23487320+weiji14@users.noreply.github.com>
     7  Michael Grund <michael_grund@gmx.de>
     6  dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
     3  Liam Toney <38269494+liamtoney@users.noreply.github.com>
     3  Meghan Jones <meghanj@hawaii.edu>
     3  Wei Ji <weiji.leong@vuw.ac.nz>
```
and so the order would be:
```suggestion
* `Will Schlitzer <https://github.com/willschlitzer>`__
* `Dongdong Tian <https://github.com/seisman>`__
* `Wei Ji Leong <https://github.com/weiji14>`__
* `Michael Grund <https://github.com/michaelgrund>`__
* `Liam Toney <https://github.com/liamtoney>`__
* `Meghan Jones <https://github.com/meghanrjones>`__
```

Comment (seisman) [reply doc/changes.rst:101]: Looks better to me.

File: doc/changes.rst
Comment (seisman) [reply doc/changes.rst:13]: > I'm also thinking whether to mention the upcoming drop in support for GMT 6.1.1 (once GMT 6.2.0 is out, something for PyGMT v0.4.0) and/or NEP29 adoption

Don't try to predict the future.

Comment (weiji14) [reply doc/changes.rst:13]: Ok, will surprise our users then :smile_cat:

File: doc/changes.rst
Comment (weiji14) [reply doc/changes.rst:101]: Alright, I'll commit the change, but we should document this (order by number of commits) for when we deal with #885.

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst]: Roughly group the changes into categories (CI, source codes, documentation, tests, and misc). The changes in the 'misc' category may be moved into "Enhancements" instead.

**NOTE**: Need to remove the blank lines before committing the changes.

```suggestion
* Add dependabot to keep GitHub Actions up to date (#861)
* Skip workflows in PRs if only non-source-code files are changed (#839)
* Add slash command '/test-gmt-dev' to test GMT dev version (#831)
* Check files for UNIX-style line breaks and 644 permission (#736)
* Rename vercel configuration file from now.json to vercel.json (#738)
* Add a CI job testing GMT master branch on Windows (#756)
* Migrate documentation deployment from Travis CI to GitHub Actions (#713)

* Move Figure.meca() into a standalone module (#686)
* Move plotting functions to separate modules (#808)
* Move non-plotting modules to separate modules (#832)
* Add isort to sort imports alphabetically (#745)
* Convert relative imports to absolute imports (#754)
* Switch from versioneer to setuptools-scm (#695)

* Add authorship policy (#726)
* Add docformatter to format plain text in docstrings (#642)
* Update CONTRIBUTING.md with formatting conventions (#775)

* Migrate pytest configurations to pyproject.toml (#725)
* Migrate coverage configurations to pyproject.toml (#667)
* Show test execution times in pytest (#835)
* Add tests for grdfilter (#809)
* Add tests for GMTInvalidInput of Figure.savefig() and Figure.show() (#810)
* Add args_in_kwargs function (#791)
* Add a Makefile target 'distclean' for deleting project metadata files (#744)
* Add a test Figure.basemap() map_scale (#739)

* Improve the error message when PyGMT fails to load the GMT library (#814)
* Add GMTInvalidInput Error for Figure.coast (#787)
* Use args_in_kwargs for basemap error raising (#797)
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:95]: Roughly group the changes into categories (CI, source codes, documentation, tests, and misc), and update description of a few changes. The changes in the 'misc' category (the last three entries) may be moved into "Enhancements" instead.

**NOTE**: Need to remove the blank lines before committing the changes.

```suggestion
* Add dependabot to keep GitHub Actions up to date (#861)
* Skip workflows in PRs if only non-source-code files are changed (#839)
* Add slash command '/test-gmt-dev' to test GMT dev version (#831)
* Check files for UNIX-style line breaks and 644 permission (#736)
* Rename vercel configuration file from now.json to vercel.json (#738)
* Add a CI job testing GMT master branch on Windows (#756)
* Migrate documentation deployment from Travis CI to GitHub Actions (#713)

* Move Figure.meca() into a standalone module (#686)
* Move plotting functions to separate modules (#808)
* Move non-plotting modules to separate modules (#832)
* Add isort to sort imports alphabetically (#745)
* Convert relative imports to absolute imports (#754)
* Switch from versioneer to setuptools-scm (#695)

* Add authorship policy (#726)
* Add docformatter to format plain text in docstrings (#642)

* Migrate pytest configurations to pyproject.toml (#725)
* Migrate coverage configurations to pyproject.toml (#667)
* Show test execution times in pytest (#835)
* Add tests for grdfilter (#809)
* Add tests for GMTInvalidInput of Figure.savefig() and Figure.show() (#810)
* Add args_in_kwargs function (#791)
* Add a Makefile target 'distclean' for deleting project metadata files (#744)
* Add a test Figure.basemap() map_scale (#739)

* Improve the error message when PyGMT fails to load the GMT library (#814)
* Add GMTInvalidInput Error for Figure.coast (#787)
* Use args_in_kwargs for basemap error raising (#797)
```

File: doc/changes.rst
Comment (seisman) [reply doc/changes.rst]: See the new comment https://github.com/GenericMappingTools/pygmt/pull/887#discussion_r575930143.

File: doc/changes.rst
Comment (weiji14) [reply doc/changes.rst:95]: I think the maintenance/documentation PR 726 could probably fit under the actual documentation section, but keep 642 in maintenance. Agree that 814 and 787 could be under enhancements, but maybe not 787 (keep it under maintenance) since that is non-user facing.

Feel free to make the change, I'm gonna have dinner first.

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:32]:
```suggestion
* Add authorship policy (#726)
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:29]:
```suggestion
* Add the shading parameter to Figure.colorbar (#752)
* Improve the error message when PyGMT fails to load the GMT library (#814)
* Add GMTInvalidInput Error for Figure.coast (#787)
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:36]: Consistently use `Figure.function()` (with parentheses） or `Figure.function` (without parentheses). 
```suggestion
* Add aliases dcw (#765) and lakes (#781) to Figure.coast()
* Add alias shading to Figure.colorbar() (#752)
* Add alias annotation (A) to Figure.contour() (#883)
* Wrap Figure.grdinfo() aliases (#799)
* Add aliases frame and cmap to Figure.colorbar() (#709)
* Add alias frame to Figure.grdview() (#707)
* Improve the error message when PyGMT fails to load the GMT library (#814)
* Add GMTInvalidInput error to Figure.coast() (#787)
```

File: doc/changes.rst
Comment (seisman) [inline doc/changes.rst:59]:
```suggestion
* Choose the best figures in tutorials for thumbnails (#826)
```

File: doc/changes.rst
Comment (weiji14) [reply doc/changes.rst:36]: I think without parentheses is better.

Comment (seisman) [reply doc/changes.rst:36]: I'm OK with both. Feel free to make any changes.

File: doc/changes.rst
Comment (seisman) [reply doc/changes.rst]: Running `git shortlog HEAD...v0.2.1 -sne` on the master branch gives me:
```
    54  Dongdong Tian <seisman.info@gmail.com>
    54  Will Schlitzer <schlitzer90@gmail.com>
     9  Wei Ji <23487320+weiji14@users.noreply.github.com>
     7  Michael Grund <michael_grund@gmx.de>
     6  dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
     3  Liam Toney <38269494+liamtoney@users.noreply.github.com>
     3  Meghan Jones <meghanj@hawaii.edu>
```

Not trying to be listed at the top, just curious why our results are different 😂

Comment (weiji14) [reply doc/changes.rst]: Hmm, I was probably on a different branch (which explains why it had my unversity vuw.ac.nz email). I think you'll get bumped down after @willschlitzer merges this PR though :rofl:

Comment (willschlitzer) [reply doc/changes.rst]: Dare I suggest that as a GMT developer and long-time PyGMT maintainer, I think @seisman has definitely earned the first spot!

File: README.rst
Comment (willschlitzer) [reply README.rst:162]: @weiji14 If you don't hear from him, his ORCID page lists "Innoplexia GmbH: Heidelberg, Baden-Württemberg, DE" as his affiliation

Comment (weiji14) [reply README.rst:162]: Ok!

Comment (michaelgrund) [reply README.rst:162]: @weiji14 I updated my affiliation, sorry for being late :sweat_smile: .

File: doc/changes.rst
Comment (weiji14) [reply doc/changes.rst]: There's still a lot of parentheses in the changelog @willschlitzer, could you remove all of them (just the v0.3.0 ones) to be consistent?

**Comment by weiji14** [comment]
> Format Pull Request links using sed -i.bak -E 's$\(#([0-9]*)\)$(`#\1 <https://github.com/GenericMappingTools/pygmt/pull/\1>`__)$g' changes.rst

Also, are you able to run `sed`? If not I can do it for you.

**Comment by willschlitzer** [comment]
> > Format Pull Request links using sed -i.bak -E 's$(#([0-9]*))$(`#\1 <https://github.com/GenericMappingTools/pygmt/pull/\1>`__)$g' changes.rst
> 
> Also, are you able to run `sed`? If not I can do it for you.

Forgot about that one! Just ran it and pushed the change

File: doc/changes.rst
Comment (weiji14) [inline doc/changes.rst:22]: Seems like the `sed` command missed a few, looks like we forgot to update it after https://github.com/GenericMappingTools/pygmt/pull/683#discussion_r523388058. Try
```
sed -i.bak -E 's$\(#([0-9]*)\)$(`#\1 <https://github.com/GenericMappingTools/pygmt/pull/\1>`__)$g' changes.rst
```

File: doc/changes.rst
Comment (weiji14) [inline doc/changes.rst:49]:
```suggestion
* Add a gallery example for Figure.logo (`#823 <https://github.com/GenericMappingTools/pygmt/pull/823>`__)
```

Can you remove all the parentheses as mentioned in https://github.com/GenericMappingTools/pygmt/pull/887#discussion_r576008258?

**Review by weiji14 — COMMENTED** [review]
Sorry, noticed a few more missing stuff at https://pygmt-git-v030-release.gmt.vercel.app/changes.html#release-v0-3-0-2021-02-15.

File: doc/changes.rst
Comment (willschlitzer) [reply doc/changes.rst]: Manually switched the formatting for some of the PR refs; the problem was when they were both in a set of parentheses.

File: doc/changes.rst
Comment (willschlitzer) [reply doc/changes.rst]: Fixed it in 98097bc

File: doc/changes.rst
Comment (weiji14) [inline doc/changes.rst:22]:
```suggestion
* Provide basic support for FreeBSD (`#700 <https://github.com/GenericMappingTools/pygmt/pull/700>`__, `#878 <https://github.com/GenericMappingTools/pygmt/pull/878>`__)
```

File: doc/changes.rst
Comment (weiji14) [inline doc/changes.rst:53]:
```suggestion
* Add more gallery examples for projections (`#761 <https://github.com/GenericMappingTools/pygmt/pull/761>`__, `#721 <https://github.com/GenericMappingTools/pygmt/pull/721>`__, `#757 <https://github.com/GenericMappingTools/pygmt/pull/757>`__, `#723 <https://github.com/GenericMappingTools/pygmt/pull/723>`__, `#762 <https://github.com/GenericMappingTools/pygmt/pull/762>`__, `#742 <https://github.com/GenericMappingTools/pygmt/pull/742>`__, `#728 <https://github.com/GenericMappingTools/pygmt/pull/728>`__, `#727 <https://github.com/GenericMappingTools/pygmt/pull/727>`__)
```

**Review by weiji14 — CHANGES_REQUESTED** [review]
Ok, manually updating two lines for now to be comma (,) separated, because I can't be bothered to learn `sed` in a rush. Hopefully this is the last :crossed_fingers:

**Comment by willschlitzer** [comment]
> Ok, manually updating two lines for now to be comma (,) separated, because I can't be bothered to learn `sed` in a rush. Hopefully this is the last crossed_fingers

Hoping for that too!

File: doc/changes.rst
Comment (weiji14) [inline doc/changes.rst:47]:
```suggestion
* Add a tutorial for the region argument (`#800 <https://github.com/GenericMappingTools/pygmt/pull/800>`__)
```
Apologies, spoke too soon. I'll commit this and spare you the torture.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**
